File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def is_prime(num):
    if num == 0 or num == 1:
        return False
    else:
        for i in range(2, (num//2)+1):
            if num % i == 0:
                return False
        return True

def solution(numbers):
    answer = 0
    pre_list = []
    num_list = list(map(int, numbers))
    print(num_list)
    
    for i in range(1, len(numbers)+1):
        for j in permutations(numbers, i):
            pre_list.append(''.join(j))
    pre_list = list(map(int, pre_list))
    pre_list = list(set(pre_list))
    
    for i in pre_list:
        if is_prime(i):
            answer += 1
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <set>
using namespace std;

set<int> numberSet;

bool isPrime(int number) {
    if(number <= 1) {
        return false;
    }
    
    for(int i=2; i*i<=number; i++) {
        if(number % i == 0) {
            return false;
        }
    }
    return true;
}

void makeCombination(string comb, string others) {
    if(comb != "") {
        numberSet.insert(stoi(comb));
    }
    for(int i=0; i<others.size(); i++) {
        makeCombination(comb + others[i], others.substr(0, i) + others.substr(i+1));
    }
}

int solution(string numbers) {
    makeCombination("", numbers);
    
    int answer = 0;
    for(int number : numberSet) {
        if(isPrime(number)) {
            answer++;
        }
    }
    return answer;
}